lib/kumascript/api.js

```diff
@@ -340,30 +340,34 @@ var APIContext = ks_utils.Class({
         return output;
     },
 
-    // #### require(path)
+    // #### require_macro(name)
     //
-    // Attempts to load and execute a template which, as a side effect, can
-    // populate an exports object in quasi-CommonJS style. The template output
-    // is ignored.
-    require: function (name) {
-
-        // Use an internal cache, so that repeated require() calls reuse the
-        // previously loaded results.
+    // This is a request to load a Kumascript macro as a module. It attempts
+    // to load and execute the named macro which, as a side effect, should
+    // populate an "exports" or "module.exports" object in nodejs style (the
+    // macro output is ignored).
+    require_macro: function (name) {
+        // Use an internal cache, so that repeated require() calls
+        // reuse the previously loaded results.
         if (!(name in this._require_cache)) {
             var clone_ctx = _.clone(this);
 
-            // Let's pretend we're following CommonJS module conventions
+            // Let's pretend we're following nodejs module conventions.
             clone_ctx.module = { exports: {} };
             clone_ctx.exports = clone_ctx.module.exports;
 
-            // clone_ctx is just like calling a template, only we ignore the output
-            // and return the side effect of populating exports.
+            // We ignore the output and return the side effect of
+            // populating the "module.exports" object.
             clone_ctx.template(name, []);
             this._require_cache[name] = clone_ctx.module.exports;
         }
-
         return this._require_cache[name];
-    }
+    },
+
+    // #### require(name)
+    //
+    // Load an npm package (the real "require" has its own cache).
+    require: require
```

Comment:
LOL, yeah this is one of the strangest lines of code I've written. It made me pause.